odoo/addons/base/models/ir_model.py

```diff
@@ -487,9 +487,10 @@ class IrModelFields(models.Model):
     state = fields.Selection([('manual', 'Custom Field'), ('base', 'Base Field')], string='Type', default='manual', required=True, readonly=True, index=True)
     on_delete = fields.Selection([('cascade', 'Cascade'), ('set null', 'Set NULL'), ('restrict', 'Restrict')],
                                  string='On Delete', default='set null', help='On delete property for many2one fields')
-    domain = fields.Char(default="[]", help="The optional domain to restrict possible values for relationship fields, "
-                                            "specified as a Python expression defining a list of triplets. "
-                                            "For example: [('color','=','red')]")
+    domain = fields.Domain(domain_model_field="model", required=False,
```

Comment:
```suggestion
    domain = fields.Domain(domain_model_field="relation", required=False,
```
and the `required=False` is clearly confusing.